arangod/ClusterEngine/ClusterTransactionState.cpp

```diff
@@ -83,6 +83,35 @@ Result ClusterTransactionState::beginTransaction(transaction::Hints hints) {
 
   transaction::ManagerFeature::manager()->registerTransaction(id(), isReadOnlyTransaction(), false /* isFollowerTransaction */);
   setRegistered();
+  
+  // simon: deadlock-mitiagtion, still unsafe in general
+  // we have to include write-level transactions too for correctness
```

Comment:
I think this code is actually correct, but has a chance to get into a now limited dead-lock situation.
And this only happens if you mix Exclusive Transactions and Write Transactions on the same collection.
If you only use one of them it works without a chance for dead-locking and the write transactions being lazy and potentially faster.

Reply:
that was the point